src/test/java/formflow/library/controllers/ScreenControllerTest.java

```diff
@@ -47,6 +49,90 @@ public void setUp() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.post("/flow/{flow}/{screen}", "testFlow", "screenThatDoesNotExist"))
+        .andExpect(status().isNotFound());
+  }
+  
+  @Test
+  void postToSubmitScreenShouldReturn404WhenFlowDoesNotExist() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.post("/flow/{flow}/{screen}/submit", "flowThatDoesNotExist", "screen"))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  void postToSubmitScreenShouldReturn404WhenScreenDoesNotExist() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.post("/flow/{flow}/{screen}/submit", "testFlow", "ScreenThatDesNotExist"))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  void getScreenInSubflowShouldReturn404WhenFlowDoesNotExist() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.get("/flow/{flow}/{screen}/thisIsAUUIDForASubflow", "flowThatDoesNotExist", "screen"))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  void getScreenInSubflowShouldReturn404WhenScreenDoesNotExist() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.get("/flow/{flow}/{screen}/thisIsAUUIDForASubflow", "testFlow", "screenThatDoesNotExist"))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  void postScreenInSubflowShouldReturn404WhenFlowDoesNotExist() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.post("/flow/{flow}/{screen}/thisIsAUUIDForASubflow", "flowThatDoesNotExist", "screen"))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  void postScreenInSubflowShouldReturn404WhenScreenDoesNotExist() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.post("/flow/{flow}/{screen}/thisIsAUUIDForASubflow", "testFlow", "screenThatDoesNotExist"))
+        .andExpect(status().isNotFound());
+  }
+  
+  @Test
+  void getToSubflowDeleteConfirmationScreenShouldReturn404WhenFlowDoesNotExist() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.get("/flow/{flow}/{subflowName}/thisIsAUUIDForASubflow/deleteConfirmation", "flowThatDoesNotExist", "testSubflow"))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  void postToSubflowDeleteShouldReturn404WhenFlowDoesNotExist() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.post("/flow/{flow}/{subflowName}/thisIsAUUIDForASubflow/delete", "flowThatDoesNotExist", "testSubflow"))
+        .andExpect(status().isNotFound());
+  }
+  
+  @Test
+  void getToScreenNavigationEndpointShouldReturn404WhenFlowDoesNotExist () throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.get("/flow/{flow}/{screen}/navigation", "flowThatDoesNotExist", "screen"))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  void getToScreenNavigationEndpointShouldReturn404WhenScreenDoesNotExist () throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.get("/flow/{flow}/{screen}/navigation", "testFlow", "screenThatDoesNotExist"))
+        .andExpect(status().isNotFound());
+  }
+
```

Comment:
It'd be great for this to be two parameterized tests (one for posts and another for gets) with `@CsvSource` for the arguments since we're using the same methods for all of them, but it's not strictly necessary.